### PR TITLE
Remove redundant queries in InstructorFeedbacksPageAction #4264

### LINKED ISSUE #4264: Remove redundant queries in InstructorFeedbacksPageAction
As title

### PR DESCRIPTION
**Pull request by imouto1994**
Fixed #4264

### CODE REVIEW

File: src/main/java/teammates/logic/core/CoursesLogic.java [hunk 1]
```diff
@@ -182,22 +184,28 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
     }
 
     public List<String> getSectionsNameForCourse(String courseId) throws EntityDoesNotExistException {
+        return getSectionsNameForCourse(courseId, true);
+    }
 
-        verifyCourseIsPresent(courseId);
-        
+    private List<String> getSectionsNameForCourse(
+        String courseId, boolean checkExists) throws EntityDoesNotExistException {
```
Comment (damithc) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: boolean variables should be named to sound like booleans e.g. `is...` `has...`

Comment (imouto1994) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: Should I rename as `hasPresentCheck` ?

File: src/main/java/teammates/logic/core/CoursesLogic.java [hunk 2]
```diff
@@ -182,22 +184,28 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
...
+        if (checkExists) {
+            verifyCourseIsPresent(courseId);
```
Comment (LowWeiLin) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: You can change the function parameter to take in CourseAttributes and assert it is not null.
There won't be a need to call verifyCourseIsPresent then.

File: src/main/java/teammates/logic/core/CoursesLogic.java (hunk 1 above)
Comment (damithc) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: Not sure. I don't know the meaning of that parameter. :-) May be you can add a comment for the method?

File: src/main/java/teammates/logic/core/CoursesLogic.java (hunk 2 above)
Comment (imouto1994) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: Okay :)

Comment (imouto1994) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: After checking, some functions still need to use this function by passing only `courseId`. Maybe we can keep it as two overloaded methods ?

Comment (LowWeiLin) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: Yes can do that. Try to change the other functions to use the one with CourseAttributes if possible.

Comment (imouto1994) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: I have created the overload methods for now. I think we can create another issue to change other functions to use the one with `CourseAttributes` object.

File: src/main/java/teammates/logic/core/CoursesLogic.java [hunk 3]
```diff
@@ -182,22 +184,39 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
...
+    }
+    
+    /**
+     * Get list of section names for a course with or without a need to check if the course is existent
+     * @param courseId Course ID of the course
+     * @param hasCheckIsPresent Determine whether it is necessary to check if the course exists
+     * @return list of sections names from the specified course
+     * @throws EntityDoesNotExistException
+     */
+    private List<String> getSectionsNameForCourse(String courseId, boolean hasCheckIsPresent)
```
Comment (LowWeiLin) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: Change variable name to isCourseVerified? And verifyCourseIsPresent only if it is false.

Comment (imouto1994) [inline src/main/java/teammates/logic/core/CoursesLogic.java]: Sounds good to me ! :)

**Comment by LowWeiLin** [comment]
Looks good to merge.

**Pull request merged**